app-ktor-server/src/main/kotlin/com/ustadmobile/lib/rest/ContentEntryLinkImporter.kt

```diff
@@ -0,0 +1,111 @@
+package com.ustadmobile.lib.rest
+
+import com.ustadmobile.core.account.Endpoint
+import io.github.aakira.napier.Napier
+import com.ustadmobile.core.contentjob.ContentJobManager
+import com.ustadmobile.core.contentjob.ContentPluginManager
+import com.ustadmobile.core.contentjob.MetadataResult
+import com.ustadmobile.core.contentjob.ProcessContext
+import com.ustadmobile.core.db.JobStatus
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.io.ext.getSize
+import com.ustadmobile.core.util.DiTag
+import com.ustadmobile.core.util.createTemporaryDir
+import com.ustadmobile.door.DoorUri
+import com.ustadmobile.door.ext.DoorTag
+import com.ustadmobile.lib.db.entities.ContentEntryWithLanguage
+import com.ustadmobile.lib.db.entities.ContentJob
+import com.ustadmobile.lib.db.entities.ContentJobItem
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.request.*
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.post
+import io.ktor.routing.route
+import kotlinx.coroutines.withTimeout
+import org.kodein.di.instance
+import org.kodein.di.ktor.closestDI
+import org.kodein.di.on
+import java.io.File
+import java.net.URL
+
+
+private val IMPORT_LINK_TIMEOUT_DEFAULT = (30 * 1000).toLong()
+
+fun Route.ContentEntryLinkImporter() {
+
+    route("import") {
+
+        post("validateLink") {
+            val url = call.request.queryParameters["url"]?: ""
+            val pluginManager: ContentPluginManager by closestDI().on(call).instance()
+            var metadata: MetadataResult? = null
+            try{
+                val processContext = ProcessContext(createTemporaryDir("content"), mutableMapOf())
```

Comment:
It looks like this temporary directory is never getting deleted. It should be deleted in a finally block.